### PR TITLE
[dhcp6relay] Add VRF support for upstream sockets (VLAN vrf and server_vrf)

### PR DESCRIPTION
**Pull request by AnantKishorSharma**
#### Why I did it
`dhcp6relay` always sent relay-forward to the DHCPv6 servers through the global routing table, so it could not reach servers that live in a non-default VRF. Fixes sonic-net/sonic-dhcp-relay#89.

#### How I did it
Bind the upstream (server-facing) socket to a VRF via `SO_BINDTODEVICE`:
- **Option A** — a relay VLAN placed in a non-default VRF (`VLAN_INTERFACE` `vrf_name`): the per-VLAN upstream socket is bound to that VRF.
- **Option B** — an explicit `server_vrf` on the `DHCP_RELAY` row (servers reachable in a VRF different from the VLAN's): a shared per-VRF upstream socket is opened, demultiplexed back to the owning VLAN by link-address. The shared socket is released once the last VLAN referencing it is removed/re-pointed, so a deleted/recreated server VRF leaves no stale socket behind.

All applied at runtime through the existing reconcile path — no container restart. A runtime VRF change tears the relay down and re-arms it — `SO_BINDTODEVICE` cannot be rebound in place, so the upstream sockets are recreated under the new VRF.

#### How to verify it
- 68/68 unit tests pass (`make test`), including new coverage for the server-VRF socket lifecycle.
- On a qfx5200 t0 hardware testbed: the existing v6 relay suite passes (no regression) plus new control-plane and data-plane VRF tests.

> Stacked on #116 — until #116 merges this PR's diff also shows #116's commit; it will be rebased to a clean diff afterwards.

### CODE REVIEW

**Comment by AnantKishorSharma** [comment]
@Xichen96 @yaqiangz @jcaiMR @saiarcot895 @yxieca @kellyyeh - Please review when you have a chance.

File: dhcp6relay/src/config_interface.cpp [hunk 1]
```diff
@@ -207,3 +236,137 @@ bool check_is_lla_ready(std::string vlan) {
...
+        publish_desired_config(config_db);
+    }
+}
+
+void start_dhcp_config_monitor(int notify_fd)
+{
+    g_notify_fd = notify_fd;
+    g_stop_monitor.store(false);
+    g_monitor_thread = std::thread(config_monitor_loop);
+    g_monitor_thread.detach();
```
Comment (lucas-li-micas) [inline dhcp6relay/src/config_interface.cpp]: The monitor thread is detached but never joined.How do we guarantee a clean shutdown and avoid accessing
destroyed resources after stop_dhcp_config_monitor()?

Comment (AnantKishorSharma) [reply dhcp6relay/src/config_interface.cpp]: Thanks @lucas-li-micas! dropped the detach() and now join() the thread in stop_dhcp_config_monitor(), so it's guaranteed to have exited before teardown. The loop already polls the stop flag once per select() timeout, so join() returns within one interval.
